src/convertToObject.js

```diff
@@ -3,16 +3,42 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes styles string
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const KEYVALUE_SEPARATOR = ';';
+  const KEY_VALUE_SEPARATOR = ':';
+
+  const keyValueArr = filterStr(sourceString).split(KEYVALUE_SEPARATOR);
+  const resultObj = {};
+
+  keyValueArr.forEach(keyValue => {
+    let [key, value] = keyValue.split(KEY_VALUE_SEPARATOR);
+
+    if (value) {
+      key = filterStr(key); value = clearValue(filterStr(value));
+      resultObj[key] = value;
+    }
+  });
+
+  return resultObj;
+}
+
+function filterStr(str) {
+  const CHARS_TO_FILTER = ['\n'];
+  const strArr = String(str).split();
+
+  return strArr.filter(char => !CHARS_TO_FILTER.includes(char)).join().trim();
+}
+
+function clearValue(valueString) {
+  return String(valueString).split(' ').filter(value => value !== '').join(' ');
```

Comment:
I think you can use  this check `.filter(value => value)`

it means if `value === true` than we keep this value (empty string is always false)